src/docs/markdown/caddyfile/directives/header.md

```diff
@@ -37,28 +37,43 @@ header [<matcher>] [[+|-|?|>]<field> [<value>|<find>] [<replace>]] {
 	?<field> <value>
 
 	[defer]
+
+	# a match condition
+	match [status <code...> | header <field> <value>]
```

Comment:
This one's a bit strange cause it's not optional to provide a matcher, so the `[]` are sorta misplaced.

I think we should probably replace these to just `<response_matcher...>` instead, similar to how we have `<matcher>` in places where a request matcher can go. We can add a jquery blob at the top to transform those into links to https://caddyserver.com/docs/caddyfile/response-matchers

Reply:
I think that's a good idea. There are two ways in which response matchers can be used. I documented the difference.

<image src="https://github.com/user-attachments/assets/ec42795c-4299-4830-83ae-28085f7e24d0" width="400"/>

`encode` and `headers` allow inline matching, `intercept` requires creating named response matchers.